Client/Frontend/Browser/Tab Tray/Views/TabSyncCustomView.swift

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import BraveUI
+
+protocol TabSyncHeaderViewDelegate {
+  func toggleSection(_ header: TabSyncHeaderView, section: Int)
+}
+
+class TabSyncHeaderView: UITableViewHeaderFooterView, TableViewReusable {
+    
+  var delegate: TabSyncHeaderViewDelegate?
+  var section = 0
+  var isCollapsed = false {
+    didSet {
+      if oldValue == isCollapsed { return }
+      
+      let rotationAngle = isCollapsed ? -1 * (.pi / 2) : 0.0
+      arrowIconView.transform = CGAffineTransform(rotationAngle: rotationAngle)
+    }
+  }
+  
+  let imageIconView = UIImageView().then {
+    $0.contentMode = .scaleAspectFit
+    $0.tintColor = .braveLabel
+  }
+    
+  let labelStackView = UIStackView().then {
+    $0.axis = .vertical
+    $0.alignment = .leading
+    $0.setContentCompressionResistancePriority(.defaultLow, for: .horizontal)
+  }
+  
+  let titleLabel = UILabel().then {
+    $0.textColor = .braveLabel
+    $0.font = .preferredFont(forTextStyle: .footnote, weight: .semibold)
+  }
+  
+  let descriptionLabel = UILabel().then {
+    $0.textColor = .secondaryBraveLabel
+    $0.font = .preferredFont(forTextStyle: .footnote)
+  }
+  
+  let arrowIconView = UIImageView().then {
+    $0.image = UIImage(systemName: "arrowtriangle.down")
+    $0.contentMode = .scaleAspectFit
+    $0.tintColor = .braveLabel
+    $0.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
+  }
+    
+  override init(reuseIdentifier: String?) {
+    super.init(reuseIdentifier: reuseIdentifier)
+    contentView.backgroundColor = .clear
+    
+    contentView.addSubview(imageIconView)
+    contentView.addSubview(labelStackView)
+    contentView.addSubview(arrowIconView)
+
+    labelStackView.addArrangedSubview(titleLabel)
+    labelStackView.setCustomSpacing(3.0, after: titleLabel)
+    labelStackView.addArrangedSubview(descriptionLabel)
+
+    imageIconView.snp.makeConstraints {
+      $0.leading.equalToSuperview().inset(TwoLineCellUX.borderViewMargin)
+      $0.centerY.equalToSuperview()
+      $0.size.equalTo(TwoLineCellUX.imageSize)
+    }
+    
+    labelStackView.snp.makeConstraints {
+      $0.leading.equalTo(imageIconView.snp.trailing).offset(TwoLineCellUX.borderViewMargin)
+      $0.trailing.equalToSuperview().offset(5.0)
```

Comment:
I think you want -5 offset here, I believe this is causing title of tabs with a long title to be slightly off-screen (final dot of ellipsis can be off-screen).

Reply:
Fixed at https://github.com/brave/brave-ios/pull/5809/commits/6ffacee00b2a7dd82abd754ee0ef50bce120f93b